hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -1769,7 +1771,7 @@ public void setTimeoutForWriteLock(long timeoutForWriteLock) {
           storeCloserThreadPool.shutdownNow();
         }
       }
-
+      storeFileOpenAndCloseThreadPool.shutdownNow();
```

Comment:
This means keeping this 'storeFileOpenAndCloseThreadPool' active till the Region is closed. 
previously we create the pool at the region open time and once the stores are opened, the pool is shutdown. Keeping it longer this way not looks good

Reply:
> This means keeping this 'storeFileOpenAndCloseThreadPool' active till the Region is closed.
> previously we create the pool at the region open time and once the stores are opened, the pool is shutdown. Keeping it longer this way not looks good

There is a timeout for this thread pool, after some time, the threads will be destroyed.

Reply:
The non core threads will get removed after the idle timeout.  But the way we create the pool, we pass the number of threads (as per the config) as the core threads count.
But we set
boundedCachedThreadPool.allowCoreThreadTimeOut(true)
So this helps. I missed this part.